src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Core;
+using System;
+using System.Linq;
+
+namespace Steeltoe.Extensions.Logging.DynamicSerilog
+{
+    public static class SerilogLoggingBuilderExtensions
+    {
+        /// <summary>
+        /// Add Steeltoe logger wrapped in a <see cref="IDynamicLoggerProvider"/> that supports
+        /// dynamically controlling the minimum log level via management endpoints
+        /// </summary>
+        /// <param name="builder">The <see cref="ILoggingBuilder"/> for configuring the LoggerFactory  </param>
+        /// <returns>The configured <see cref="ILoggingBuilder"/></returns>
+        public static ILoggingBuilder AddSerilogDynamicConsole(this ILoggingBuilder builder)
+        {
+            return builder.AddSerilogDynamicConsole(null, false);
+        }
+
+        /// <summary>
+        /// Add Steeltoe logger wrapped in a <see cref="IDynamicLoggerProvider"/> that supports
+        /// dynamically controlling the minimum log level via management endpoints
+        /// </summary>
+        /// <param name="builder">The <see cref="ILoggingBuilder"/> for configuring the LoggerFactory  </param>
+        /// <param name="loggerConfiguration">An initial <see cref="LoggerConfiguration"/></param>
+        /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
+        /// <returns>The configured <see cref="ILoggingBuilder"/></returns>
+        public static ILoggingBuilder AddSerilogDynamicConsole(this ILoggingBuilder builder, LoggerConfiguration loggerConfiguration, bool preserveStaticLogger = false)
+        {
+            if (builder.Services.Any(sd => sd.ServiceType == typeof(IDynamicLoggerProvider)))
+            {
+                throw new InvalidOperationException("An IDynamicLoggerProvider has already been configured! Call 'AddSerilogDynamicConsole' earlier in program.cs (before adding Actuators) or remove duplicate IDynamicLoggerProvider entries.");
```

Comment:
(deja vu?) The exclamation point seems redundant.

Reply:
This isn't new code, it just looks like it because the file rename shows up as a delete/add... this?
```suggestion
                throw new InvalidOperationException("An IDynamicLoggerProvider has already been configured. Call 'AddSerilogDynamicConsole' earlier in program.cs (before adding Actuators) or remove duplicate IDynamicLoggerProvider entries.");
```

Reply:
Looks good